saq/queue/postgres.py

```diff
@@ -110,31 +110,34 @@ def __init__(
         self.connection_lock = asyncio.Lock()
         self.released: list[str] = []
         self.has_sweep_lock = False
-        self.tasks: list[asyncio.Task] = []
 
-    async def connect(self) -> None:
-        if self.connection:
-            # If connection exists, connect() was already called
-            return
-
-        await self.pool.open()
-        await self.pool.resize(min_size=self.min_size, max_size=self.max_size)
+    async def init_db(self) -> None:
         async with self.pool.connection() as conn, conn.cursor() as cursor:
             for statement in DDL_STATEMENTS:
                 await cursor.execute(
                     SQL(statement).format(
                         jobs_table=self.jobs_table, stats_table=self.stats_table
                     )
                 )
-        # Reserve a connection for dequeue and advisory locks
-        self.connection = await self.pool.getconn()
 
-        self.tasks.append(asyncio.create_task(self.wait_for_job()))
-        self.tasks.append(asyncio.create_task(self.listen_for_enqueues()))
+    async def upkeep(self) -> set[asyncio.Task[None]]:
+        await self.init_db()
+
+        self.tasks.add(asyncio.create_task(self.wait_for_job()))
```

Comment:
it doesn't look like queue.tasks is used anymore, can we remove it? or do we actually need to use it?

Reply:
I gave Queue its own stop() method.